### PR TITLE
Support national clouds in authentication layer

### PR DESCRIPTION
**Pull request by Ndiritu**
part of https://github.com/microsoft/kiota-authentication-phpleague-php/issues/35

depends on https://github.com/microsoft/kiota-authentication-phpleague-php/pull/37 being released

### CODE REVIEW

File: src/Authentication/GraphPhpLeagueAccessTokenProvider.php [hunk 1]
```diff
@@ -24,14 +27,37 @@
  */
 class GraphPhpLeagueAccessTokenProvider extends PhpLeagueAccessTokenProvider
 {
+    public const NATIONAL_CLOUD_TO_AZURE_AD_ENDPOINT = [
+        NationalCloud::GLOBAL => 'https://login.microsoftonline.com',
+        NationalCloud::US_GOV => 'https://login.microsoftonline.us',
+        NationalCloud::CHINA => 'https://login.chinacloudapi.cn'
```
Comment (andrueastman) [inline src/Authentication/GraphPhpLeagueAccessTokenProvider.php:33]: I think we should also add the Germany one as the Azure folks have. 
https://github.com/Azure/azure-sdk-for-net/blob/1d3d91cf0dbe0c71cc1fa3753ca284674b19a513/sdk/identity/Azure.Identity/src/AzureAuthorityHosts.cs#L15

File: src/Authentication/GraphPhpLeagueAccessTokenProvider.php [hunk 2]
```diff
@@ -24,14 +27,37 @@
...
+     * https://learn.microsoft.com/en-us/graph/deployments
      */
-    public function __construct(TokenRequestContext $tokenRequestContext, array $scopes = [])
+    public function __construct(
+        TokenRequestContext $tokenRequestContext,
+        array $scopes = [],
+        string $nationalCloud = NationalCloud::GLOBAL
+    )
     {
         $allowedHosts = ["graph.microsoft.com", "graph.microsoft.us", "dod-graph.microsoft.us", "graph.microsoft.de",
```
Comment (andrueastman) [inline src/Authentication/GraphPhpLeagueAccessTokenProvider.php:48]: We should also `graph.microsoft-ppe.com` in the list

File: src/Authentication/GraphPhpLeagueAccessTokenProvider.php (hunk 1 above)
Comment (Ndiritu) [reply src/Authentication/GraphPhpLeagueAccessTokenProvider.php:33]: Looks like it was deprecated - https://github.com/microsoftgraph/microsoft-graph-docs/pull/17947

File: src/Authentication/GraphPhpLeagueAccessTokenProvider.php (hunk 2 above)
Comment (Ndiritu) [reply src/Authentication/GraphPhpLeagueAccessTokenProvider.php]: Added.

File: src/Authentication/GraphPhpLeagueAccessTokenProvider.php (hunk 1 above)
Comment (andrueastman) [reply src/Authentication/GraphPhpLeagueAccessTokenProvider.php:33]: Happy to keep it as is then. 👍🏼

**Review by andrueastman — APPROVED** [review]

**Pull request merged**